blockchain/state_manager/src/lib.rs

```diff
@@ -359,13 +358,15 @@ where
                 &rand_clone,
                 base_fee.clone(),
                 network_version,
-                |height, state| self.genesis_info.get_circulating_supply(height, state),
+                |height, db| {
+                    let state = StateTree::new_from_root(db, &state_root)?;
+                    self.genesis_info.get_circulating_supply(height, &state)
```

Comment:
Could you move `new_from_root` into `get_circulating_supply`?

Reply:
Done.